src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -146,40 +146,6 @@ public override IReadOnlyList<Location> AdditionalLocations
                 get { return _properties; }
             }
 
-            public override bool Equals(Diagnostic obj)
-            {
-                var other = obj as SimpleDiagnostic;
-                if (other == null)
-                {
-                    return false;
-                }
-
-                if (AnalyzerExecutor.IsAnalyzerExceptionDiagnostic(this))
-                {
-                    // We have custom Equals logic for diagnostics generated for analyzer exceptions.
-                    return AnalyzerExecutor.AreEquivalentAnalyzerExceptionDiagnostics(this, other);
```

Comment:
You will need an equivalent de-duplication logic for analyzer exception diagnostics at least here in the command line compiler: http://source.roslyn.io/#Microsoft.CodeAnalysis/CommandLine/CommonCompiler.cs,367. Otherwise you will see lots of duplicate diagnostics reported for an analyzer throwing identical exception for different callbacks.